### ISSUE TITLE
no-unused-prop-types false positive with shared `Props`

### ISSUE THREAD

**Issue by eventualbuddha** [yannickcr/eslint-plugin-react#3050]
In [this file](https://github.com/votingworks/vxsuite/blob/0a91c966b761782400a361ed36ad947605b76588/apps/precinct-scanner/src/components/Layout.tsx) `no-unused-prop-types` flags `infoBar` and `infoBarMode` as violating the rule. I believe this is because in _one_ of the components within the file they are not used. However, I believe this is a false positive because in the other component they are used:

```ts
import React from 'react'
import { Screen, Main, MainChild, Prose, fontSizeTheme } from '@votingworks/ui'
import ElectionInfoBar, { InfoBarMode } from './ElectionInfoBar'

interface Props {
  children?: React.ReactNode
  infoBar?: boolean
  infoBarMode?: InfoBarMode
}

export const CenteredScreen = ({
  children,
  infoBar = true,
  infoBarMode,
}: Props): JSX.Element => (
  <Screen flexDirection="column">
    <Main padded>
      <MainChild center maxWidth={false}>
        {children}
      </MainChild>
    </Main>
    {infoBar && <ElectionInfoBar mode={infoBarMode} />}
  </Screen>
)

export const CenteredLargeProse = ({ children }: Props): JSX.Element => (
  <Prose textCenter maxWidth={false} theme={fontSizeTheme.large}>
    {children}
  </Prose>
)
```

Or, a reduced test case:

```ts
import React from 'react'

interface Props {
  a?: unknown
  b?: unknown
  c?: unknown
}

export const A = ({ a, b, c }: Props): JSX.Element => (
  <div>
    {typeof a} {typeof b} {typeof c}
  </div>
)

export const B = ({ c }: Props): JSX.Element => <div>{typeof c}</div>
```

In this reduced test case `a` and `b` are improperly flagged.

**Comment by ljharb** [comment]
Closing, but https://github.com/yannickcr/eslint-plugin-react/issues/3050#issuecomment-905961256 remains true.
